csharp/src/Apache.Arrow/Memory/NullMemoryOwner.cs

```diff
@@ -26,4 +26,13 @@ public void Dispose()
         {
         }
     }
-}
\ No newline at end of file
+
+    internal class LargeNullMemoryOwner: ILargeMemoryOwner<byte>
```

Comment:
Instead of a new class, we can make `NullMemoryOwner` also implement `ILargeMemoryOwner<byte>`